config/m4/cuda.m4

```diff
@@ -107,7 +107,9 @@ AS_IF([test "x$cuda_checked" != "xyes"],
          AS_IF([test "x$cuda_happy" = "xyes"],
                [AS_IF([test "x$with_nvcc_gencode" = "xdefault"],
                       [AS_IF([test $CUDA_MAJOR_VERSION -eq 11],
-                             [NVCC_ARCH="${ARCH7} ${ARCH8} ${ARCH9} ${ARCH10} ${ARCH11} ${ARCH12}"])],
+			     [NVCC_ARCH="${ARCH7} ${ARCH8} ${ARCH9} ${ARCH10} ${ARCH11}"],
+			     [test $CUDA_MAJOR_VERSION -eq 12],
```

Comment:
I think this should be CUDA 11.1, not CUDA 12. To double check, you can run
```
nvcc --version &&  touch test.cu && nvcc -gencode arch=compute_86,code=compute_86 test.cu
```
in
```
# CUDA 11.0
docker run --gpus all -it nvcr.io/nvidia/pytorch:20.08-py3
# CUDA 11.1
docker run --gpus all -it nvcr.io/nvidia/pytorch:20.12-py3
```

Reply:
@zasdfgbnm - Thanks for the info, I likely made the wrong assumption. I just double-checked, the flag did exist in 11.4, but not in 11.0. Is it introduced in 11.1 I presume? I will update the PR to check the minor version then.

Reply:
@zasdfgbnm @Sergei-Lebedev - One side question, I thought `ARCH12` means new flags introduced by CUDA 12, thus made the wrong assumption. Looks like `sm_86` was introduced by CUDA 11.1, would `ARCH111` be a better name to avoid confusion? WDYT?

Reply:
I prefer the `ARCH111` naming scheme :)

Reply:
@kingchc ARCH111 looks good to me, pls also rename ARCH11 to ARCH110 and others as @zasdfgbnm suggested

Reply:
Updated